Lib/weakref.py

```diff
@@ -488,6 +488,25 @@ def update(self, dict=None, /, **kwargs):
         if len(kwargs):
             self.update(kwargs)
 
+    def __ior__(self, other):
+        self.update(other)
+        return self
+
+    def __or__(self, other):
+        if isinstance(other, (dict, WeakKeyDictionary, WeakValueDictionary)):
+            c = self.copy()
+            c.update(other)
+            return c
+        return NotImplemented
+
+    def __ror__(self, other):
+        if isinstance(other, (dict, WeakKeyDictionary, WeakValueDictionary)):
```

Comment:
This seems even more doubtful -- why would WeakValueDict | WeakKeyDict return a WeakKeyDict instead of a WeakValueDict? Aren't they symmetrical? (Also there are no tests for these mixes.)

Reply:
When `__ror__` is used, should it always return the same type as `other`?

Reply:
No, it should return the type of `self`. If `other` had wanted to be the return type, that class could have done so in its `__or__` method, which is called before `__ror__`. But apparently they didn't, so it's ours.

(Here's a clarifying example for why it should be this way: Consider `1 + 2.5`. `int.__or__` returns `NotImplemented` when `other` is a `float`, so then `float.__ror__` is invoked with `other` being the `int`, and it returns the `float` value `3.5`!)